src/write_influxdb_udp.c

```diff
@@ -365,136 +365,131 @@ static int wifxudp_escape_string(char *buffer, size_t buffer_size,
-                                const data_set_t *ds, const value_list_t *vl) {
+                                metric_family_t const *fam, size_t *idx) {
   int status;
-  int offset = 0;
-  gauge_t *rates = NULL;
+  int metric_offset = 0;
   bool have_values = false;
 
-  assert(0 == strcmp(ds->type, vl->type));
-
 #define BUFFER_ADD_ESCAPE(...)                                                 \
   do {                                                                         \
     status = wifxudp_escape_string(buffer + offset, buffer_len - offset,       \
                                    __VA_ARGS__);                               \
-    if (status < 0)                                                            \
-      return -1;                                                               \
+    if (status < 0) {                                                          \
+      return metric_offset;                                                    \
+    }                                                                          \
     offset += status;                                                          \
   } while (0)
 
 #define BUFFER_ADD(...)                                                        \
   do {                                                                         \
     status = snprintf(buffer + offset, buffer_len - offset, __VA_ARGS__);      \
     if ((status < 0) || (status >= (buffer_len - offset))) {                   \
-      sfree(rates);                                                            \
-      return -1;                                                               \
+      return metric_offset;                                                    \
     }                                                                          \
     offset += status;                                                          \
   } while (0)
 
-  BUFFER_ADD_ESCAPE(vl->plugin);
-  BUFFER_ADD(",host=");
-  BUFFER_ADD_ESCAPE(vl->host);
-  if (strcmp(vl->plugin_instance, "") != 0) {
-    BUFFER_ADD(",instance=");
-    BUFFER_ADD_ESCAPE(vl->plugin_instance);
-  }
-  if (strcmp(vl->type, "") != 0) {
-    BUFFER_ADD(",type=");
-    BUFFER_ADD_ESCAPE(vl->type);
-  }
-  if (strcmp(vl->type_instance, "") != 0) {
-    BUFFER_ADD(",type_instance=");
-    BUFFER_ADD_ESCAPE(vl->type_instance);
-  }
+  for (; *idx < fam->metric.num; (*idx)++) {
+    int offset = metric_offset;
+    have_values = false;
+    BUFFER_ADD_ESCAPE(fam->name);
+    metric_t metric = fam->metric.ptr[*idx];
 
-  BUFFER_ADD(" ");
-  for (size_t i = 0; i < ds->ds_num; i++) {
-    if ((ds->ds[i].type != DS_TYPE_COUNTER) &&
-        (ds->ds[i].type != DS_TYPE_GAUGE) &&
-        (ds->ds[i].type != DS_TYPE_DERIVE)) {
-      sfree(rates);
-      return -1;
+    for (size_t j = 0; j < metric.label.num; j++) {
+      label_pair_t label = metric.label.ptr[j];
+      if (strcmp(label.name, "") != 0 && strcmp(label.value, "") != 0) {
```

Comment:
Theoretically yes, but what I have found while inspecting the kstat code is that metrics with an empty name _can_ be reported. If all reader modules consistently skip empty metrics your statement is true, but is this the case?

Reply:
This comes from bad exec plugins and other software sending metrics directly to collectd with empty plugin_instance, type, etc. If it does not apply in v6 it can be removed ofc.

Reply:
That label names are not empty is ensured in `label_set_create`. An empty label value is treated as "delete this label" in `metric_label_set`. There are other ways to allocate the structure, circumventing these checks, but it should be quite rare.

On the other hand, these checks here don't really hurt either, so if you prefer we can keep them in as well.

Reply:
In the optimal case the place where empty values are removed should be defined explicitly, like "no read plugin may return empty values" and if they still do it is a bug. Introducing late checks on individual writer plugins result in behaviour deviances depending on the writer plugin which should in my opinion be avoided.

Reply:
Removed